### DISCUSSION TITLE
Spotify disconnecting immediately playback starts

### DISCUSSION THREAD

**Discussion by sillenomis** [rern/rAudio#444] [Bug Reports]
Version:rAudio 1 • 20210716
Kernel:5.10.50-1-ARCH armv6l
Hardware:Raspberry Pi Zero W Rev 1.1 & Allo Mini Boss

Setup:
Spotify enabled in rAudio, device visible from spotify app. 

Method One:
Selected spotifyd@raudio as output from spotify app, start playing music on app and app immediately disconnects from rAudio. No playback on rAudio. 

Method Two:
Started music playback on spotify app and then selected rAudio, music continues to play on app (although no sound, as if music is playing on connected device) and app remains connected, but no audio on rAudio and no indication spotify is connected. 

Notes:
Replicated issue connecting from Windows 10, IOS & Android. 
Confirmed audio is working on rAudi on playing web radio stream.

**Comment by retired-guy** [comment]
There's a known issue with this, caused by compiling the spotify daemon with the current version of Rust.  Workaround appears to be recompiling with an earlier version of Rust.  From journalctl:

Sep 07 14:02:21 rAudio spotifyd[1464]: Using alsa sink
Sep 07 14:02:21 rAudio spotifyd[1464]: Country: "US"
Sep 07 14:02:22 rAudio spotifyd[1464]: Loading <This Year's Kisses> with Spotify URI <spotify:track:2J8I2jNemsQHWhqLwTUm0q>
Sep 07 14:02:22 rAudio spotifyd[1464]: **Caught panic with message: attempted to zero-initialize type `librespot_tremor::tremor_sys::ov_callbacks`, which is invalid**
Sep 07 14:02:25 rAudio spotifyd[1464]: Caught panic with message: called `Result::unwrap()` on an `Err` value: "SendError(..)"
Sep 07 14:02:25 rAudio spotifyd[1464]: Player thread panicked!
Sep 07 14:02:25 rAudio systemd[1]: spotifyd.service: Main process exited, code=exited, status=101/n/a
Sep 07 14:02:25 rAudio systemd[1]: spotifyd.service: Failed with result 'exit-code'.
Sep 07 14:02:37 rAudio systemd[1]: spotifyd.service: Scheduled restart job, restart counter is at 1.
